.github/workflows/analysis-of-endpoint-connections.yml

```diff
@@ -24,39 +20,68 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Get list of modified files
-        run: |
-          git diff --name-only origin/${{ github.event.pull_request.base.ref }} HEAD > modified_files.txt
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '${{ env.java }}'
+          distribution: 'temurin'
 
-      # Analyze the client sided REST-API calls
-      - name: Set up Node.js
+      - name: Set up node.js
         uses: actions/setup-node@v4
         with:
           node-version: '${{ env.node }}'
```

Comment:
How many runners to we have? Would it make sense to add some caching here? I think this could speed up the runners a bit :)

Reply:
caching for node is not supported, I added caching to the gradle things though.